bugfix generalised Laplace

beauof opened the pull request:
- some variables were not initialised
- this caused unexpected behavior (e.g. NANs in element matrices) in debug builds
- closes issue #81

beauof committed:
bugfix generalised Laplace

- some variables were not initialised
- this caused unexpected behavior (e.g. NANs in element matrices) in debug builds

chrispbradley commented on src/Laplace_equations_routines.f90:
```diff
@@ -650,6 +650,10 @@ SUBROUTINE LAPLACE_EQUATION_FINITE_ELEMENT_CALCULATE(EQUATIONS_SET,ELEMENT_NUMBE
+            dXdNu=0.0_RP
+            dNudXi=0.0_RP
+            dXidNu=0.0_RP
             CALL Coordinates_MaterialSystemCalculate(GEOMETRIC_INTERP_POINT_METRICS,FIBRE_INTERPOLATED_POINT, &
```
These variables are INTENT(OUT) in Coordinates_MaterialSystemCalculate and so it shouldn't matter about them being initalised before hand? You mention that you were getting NaNs? Is that the problem? If so, then I suspect the real problem is elsewhere. The material system will only calculate the appropriate number of dimensions of these tensors i.e., if we are in 2D then only the first 2x2 entries will be set. This would leave the remaining entries of the tensor as possible NaNs. This is fine and the real problem would be that these NaNs are accessed and used. I can see that immediately below dNuDXi is used in a matrix product. I suspect the real solution would be to ensure that the correct number of dimensions in used i.e.,

CALL MatrixProduct(dNudXi(1:numberOfDimensions,1:numberOfDimensions),conductivityMaterial(1:numberOfDimensions,1:numberOfDimensions),conductivityTemp(1:numberOfDimensions,1:numberOfDimensions),err,error,*999)

etc.

chrispbradley reviewed (changes requested):
You will need to fix the dimension problem as you have the same problem with the conductivity material problem as this only fills in the top 2x2 positions for 2D.

beauof committed:
commit addresses requested changes on pull-request #82 for issue #81

- bugfix for 2D generalised Laplace code
- also conversion of routine LAPLACE_EQUATION_FINITE_ELEMENT_CALCULATE to new coding style: LaplaceEquation_FiniteElementCalculate

beauof committed:
small additional fixes for commit 26076f3

beauof commented:
Thanks @chrispbradley for reviewing the commit.

I have addressed your comments and also converted the modified routine to the new coding style.

Pull-request is ready for further review.
